core/jazz_slack-channel/test/test.js

```diff
@@ -1,11 +1,1025 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const expect = require('chai').expect;
+const awsContext = require('aws-lambda-mock-context');
+const AWS = require('aws-sdk-mock');
+const sinon = require('sinon');
+const request = require('request');
+const format = require("string-template");
+
 const index = require('../index');
+const configObj = require('../components/config.js');
+
+describe('jazz_slack-channel', function () {
+  var err, event, context, callback, callbackObj, reqStub;
+  beforeEach(function () {
+    event = {
+      "stage": "test",
+      "method": "POST",
+      "body": {
+        "channel_name" : "abc-channel",
+        "users" : [{"email_id" :"xyz@abc.com"}]
+      },
+      "principalId": "xswdxwscvff@test.com"
+    };
+    context = awsContext();
+    callback = (err, responseObj) => {
+      if (err) {
+        return err;
+      } else {
+        return JSON.stringify(responseObj);
+      }
+    };
+    err = {
+      "errorType": "svtfoe",
+      "message": "starco"
+    };
+    callbackObj = {
+      "callback": callback
+    };
+    config = configObj.getConfig(event, context);
+  });
+
+  describe('genericInputValidation', () => {
+    it("should indicate input error if service event is not defined/invalid", () => {
+      let invalidArray = ["", null, undefined];
+      invalidArray.forEach(each => {
+        event = each;
+        index.genericInputValidation(event, config)
+        .catch(error => {
+          expect(error).to.include({ result: 'inputError', message: 'Service inputs not defined!' })
+        });
+      });
+    });
+
+    it("should indicate input error if service event.body is not defined/invalid", () => {
+      let invalidArray = ["", null, undefined];
+      invalidArray.forEach(each => {
+        event.body = each;
+        index.genericInputValidation(event, config)
+        .catch(error => {
+          expect(error).to.include({ result: 'inputError', message: 'Service inputs not defined!' })
+        });
+      });
+    });
+
+    it("should indicate input error if service event.method is not defined/invalid", () => {
+      let invalidArray = ["", null, undefined];
+      invalidArray.forEach(each => {
+        event.method = each;
+        index.genericInputValidation(event, config)
+        .catch(error => {
+          expect(error).to.include({ result: 'inputError', message: 'Service inputs not defined!' })
```

Comment:
ideally we should provide a better error message which input is not defined. Service inputs not defined - 'method'
This is more of a change in the source code. What do you think?

Reply:
Changes made in source code.